crates/gdtk-ast/src/ast.rs

```diff
@@ -3,24 +3,29 @@
 
 /// A block of statements.
 pub type CodeBlock<'a> = Vec<ASTStatement<'a>>;
-pub type DictValue<'a> = Vec<(ASTExpr<'a>, ASTExpr<'a>)>;
+pub type DictValue<'a> = (ASTExpr<'a>, ASTExpr<'a>);
 pub type DictPattern<'a> = (ASTExpr<'a>, Option<Box<ASTMatchPattern<'a>>>);
 
-#[derive(Debug, Clone, PartialEq)]
+type Range = std::ops::Range<usize>;
```

Comment:
This should be moved into a separate crate for re-usability, similar to ruff's `ruff_text_size` or rustc's `rustc_span`.